standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/utils/TestMetaStoreServerUtils.java

```diff
@@ -293,6 +335,93 @@ public void testUpdateTableStatsSlow_doesNotUpdateStats() throws TException {
     MetaStoreServerUtils.updateTableStatsSlow(db, tbl2, wh, false, false, null);
     verify(wh, never()).getFileStatusesForUnpartitionedTable(db, tbl2);
   }
+  
+  /**
+   * Verify that updateTableStatsForCreateTable() does not invoke calculation of table statistics when
+   * <ol>
+   *   <li>the table is an Iceberg table</li>
+   * </ol>
+   */
+  @Test
+  public void testUpdateTableStatsForCreateTableDoesNotInvokeStatsCalc() throws TException {
+    // Iceberg table & Iceberg stats source not defined (default is Iceberg) => doesn't invoke stats calculation
+    Map<String, String> params = new HashMap<>(paramsWithStats);
+    Warehouse wh = mock(Warehouse.class);
+
+    Table tbl1 = new TableBuilder()
```

Comment:
I'm wondering if we can check the change of table's parameters to test it out, i.e, the fast stats.